patchwork-events-world/src/main/java/net/minecraftforge/event/world/WorldEvent.java

```diff
@@ -0,0 +1,198 @@
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.world;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.common.MinecraftForge;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.network.ClientPlayNetworkHandler;
+import net.minecraft.client.render.WorldRenderer;
+import net.minecraft.client.world.ClientWorld;
+import net.minecraft.entity.EntityCategory;
+import net.minecraft.server.WorldGenerationProgressListener;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.ProgressListener;
+import net.minecraft.util.profiler.Profiler;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IWorld;
+import net.minecraft.world.SpawnHelper;
+import net.minecraft.world.WorldSaveHandler;
+import net.minecraft.world.gen.chunk.ChunkGenerator;
+import net.minecraft.world.level.LevelInfo;
+import net.minecraft.world.biome.Biome.SpawnEntry;
+import net.minecraft.world.dimension.DimensionType;
+import net.minecraft.world.level.LevelProperties;
+
+/**
+ * WorldEvent is fired when an event involving the world occurs.
+ *
+ * <p>If a method utilizes this {@link Event} as its parameter, the method will
+ * receive every child event of this class.</p>
+ *
+ * <p>{@link #world} contains the World this event is occurring in.</p>
+ *
+ * <p>All children of this event are fired on the {@link MinecraftForge#EVENT_BUS}.</p>
+ */
+public class WorldEvent extends Event {
+	private final IWorld world;
+
+	// For EventBus
+	public WorldEvent() {
+		this.world = null;
+	}
+
+	public WorldEvent(IWorld world) {
+		this.world = world;
+	}
+
+	public IWorld getWorld() {
+		return world;
+	}
+
+	/**
+	 * WorldEvent.Load is fired when Minecraft loads a world.
+	 *
+	 * <p>This event is fired when a world is loaded in
+	 * {@link ClientWorld#ClientWorld(ClientPlayNetworkHandler, LevelInfo, DimensionType, int, Profiler, WorldRenderer)},
+	 * {@link MinecraftServer#createWorlds(WorldSaveHandler, LevelProperties, LevelInfo, WorldGenerationProgressListener)},
+	 * TODO: {@link DimensionManager#initDimension(int)}</p>
+	 *
+	 * <p>This event is not Cancelable.</p>
```

Comment:
Same for the other lines
```suggestion
	 * <p>This event is not cancellable.</p>
```